node-manager/src/localstorage.rs

```diff
@@ -0,0 +1,946 @@
+        let key = MapKey::LastIndex(keychain).as_map_key();
+        let res: Option<u32> = LocalStorage::get(&key).ok();
+        LocalStorage::delete(&key);
+
+        Ok(res)
+    }
+    fn del_sync_time(&mut self) -> Result<Option<SyncTime>, bdk::Error> {
+        let key = MapKey::SyncTime.as_map_key();
+        let res: Option<SyncTime> = LocalStorage::get(&key).ok();
+        LocalStorage::delete(&key);
+
+        Ok(res)
+    }
+}
+
+impl Database for MutinyBrowserStorage {
+    fn check_descriptor_checksum<B: AsRef<[u8]>>(
+        &mut self,
+        keychain: KeychainKind,
+        bytes: B,
+    ) -> Result<(), bdk::Error> {
+        let key = MapKey::DescriptorChecksum(keychain).as_map_key();
+
+        let prev = LocalStorage::get::<Vec<u8>>(&key).ok();
+        if let Some(val) = prev {
+            if val == bytes.as_ref().to_vec() {
+                Ok(())
+            } else {
+                Err(bdk::Error::ChecksumMismatch)
+            }
+        } else {
+            self.set(key, bytes.as_ref().to_vec())?;
+            Ok(())
+        }
+    }
+
+    fn iter_script_pubkeys(
+        &self,
+        keychain: Option<KeychainKind>,
+    ) -> Result<Vec<Script>, bdk::Error> {
+        let key = MapKey::Path((keychain, None)).as_map_key();
+        self.scan_prefix(key)
+            .into_iter()
+            .map(|(_, value)| -> Result<_, bdk::Error> {
+                let str = value.as_str().expect("Unexpected json value");
+                Script::from_hex(str)
+                    .map_err(|_| bdk::Error::Generic(String::from("Error decoding")))
+            })
+            .collect()
+    }
+
+    fn iter_utxos(&self) -> Result<Vec<LocalUtxo>, bdk::Error> {
+        let key = MapKey::Utxo(None).as_map_key();
+        self.scan_prefix(key)
+            .into_iter()
+            .map(|(_, value)| -> Result<_, bdk::Error> {
+                let utxo: LocalUtxo = Deserialize::deserialize(value)?;
+                Ok(utxo)
+            })
+            .collect()
+    }
+
+    fn iter_raw_txs(&self) -> Result<Vec<Transaction>, bdk::Error> {
+        let key = MapKey::RawTx(None).as_map_key();
+        self.scan_prefix(key)
+            .into_iter()
+            .map(|(_, value)| -> Result<_, bdk::Error> {
+                let tx: Transaction = Deserialize::deserialize(value)?;
+                Ok(tx)
+            })
+            .collect()
+    }
+
+    fn iter_txs(&self, include_raw: bool) -> Result<Vec<TransactionDetails>, bdk::Error> {
+        let key = MapKey::Transaction(None).as_map_key();
+        self.scan_prefix(key)
+            .into_iter()
+            .map(|(key, value)| -> Result<_, bdk::Error> {
+                let mut tx_details: TransactionDetails = Deserialize::deserialize(value)?;
+                if include_raw {
+                    // first byte is prefix for the map, need to drop it
+                    let rm_prefix: &str = &key[2..key.len()];
```

Comment:
direct array access can panic so you gotta be certain you def have this... if you're comfy with it I am, I just usually get these wrong :)

can use `get` if you want a way to handle failure: https://doc.rust-lang.org/std/primitive.slice.html#method.get

Reply:
Okay switched to use that, seems safer now